src/pubsub.c

```diff
@@ -374,7 +374,6 @@ void pubsubShardUnsubscribeAllChannelsInSlot(unsigned int slot) {
     while ((de = dictNext(di)) != NULL) {
         robj *channel = dictGetKey(de);
         dict *clients = dictGetVal(de);
-        if (dictSize(clients) == 0) goto cleanup;
```

Comment:
Why is this redundant?

If clients can never be empty here, maybe we should add an assert on size == 0 here?

Reply:
Because every time we delete a client from the client dict, we will check if the client dict is empty. If empty, we will delete it from the parent dict. So we don't have an empty dict here and don't need to check it. I think this code may make developer confused because it can never be true.

Even if the client dict isn't empty here for some reason(i.e. modify some codes later), there won't be any problem and we just iterate an empty dict. I think we don't need to add an assert here.